cpp/include/raft/neighbors/detail/cagra/cagra_build.cuh

```diff
@@ -40,6 +42,23 @@
 
 namespace raft::neighbors::cagra::detail {
 
+template <typename DataT, typename accessor>
+ivf_pq::index_params get_default_ivf_pq_build_params(
+  mdspan<const DataT, matrix_extent<int64_t>, row_major, accessor> dataset,
+  const raft::distance::DistanceType metric)
+{
+  auto build_params    = ivf_pq::index_params{};
+  build_params.n_lists = dataset.extent(0) < 4 * 2500 ? 4 : (uint32_t)(dataset.extent(0) / 2500);
+  build_params.pq_dim  = raft::Pow2<8>::roundUp(dataset.extent(1) / 2);
+  build_params.pq_bits = 8;
+  build_params.kmeans_trainset_fraction = dataset.extent(0) < 10000 ? 1 : 10;
+  build_params.kmeans_n_iters           = 25;
```

Comment:
I'm not sure if we should adjust this from the standard default. Maybe remove this and let the actual default (I believe 10) be used.